File: TrueVote.Services/Database/Pitanje.cs
using System;
using System.Collections.Generic;

namespace TrueVote.Services.Database;

public partial class Pitanje
{
    public int Id { get; set; }

    public int KategorijaId { get; set; }

    public string PitanjeText { get; set; } = null!;

    public string OdgovorText { get; set; } = null!;

    public DateTime DatumKreiranja { get; set; }

    public virtual Kategorija Kategorija { get; set; } = null!;
}

File: UI/truevote_desktop/lib/models/pitanje.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:truevote_desktop/models/kategorija.dart';
part 'pitanje.g.dart';

@JsonSerializable()
class Pitanje {
  int id;
  int? kategorijaId;
  String? pitanjeText;
  String? odgovorText;
  DateTime? datumKreiranja;
  Kategorija? kategorija;

  Pitanje({required this.id, this.kategorijaId, this.pitanjeText, this.odgovorText, this.datumKreiranja, this.kategorija});

  factory Pitanje.fromJson(Map<String, dynamic> json) =>
      _$PitanjeFromJson(json);
  Map<String, dynamic> toJson() => _$PitanjeToJson(this);
}
